### ISSUE TITLE
ResourceMap is unsafe

### ISSUE THREAD

**Issue by mbaudin47** [openturns/openturns#1141]
The following code surprisingly works:
```
import openturns as ot
ot.ResourceMap_SetAsUnsignedInteger("Jingle-Bells",0) 
```
This is unsafe: the "Jingle-Bells" key does not exist in the dictionnary. Hence, an error must be generated.

**Comment by jschueller** [comment]
We want to be able to set entries in the ResourceMap this way: modules requires that, and also some classes that do not define default values for these entries, for example the Ceres class.

**Comment by mathieu-couplet** [comment]
Hello.

I would like this ticket to be reconsidered.

First remark : a trade-off could be to give a warning in return to the user.

It is really unsafe in practice due to fat-finger errors, or the inversion of the order of string arguments (value then key) - with ResourceMap_Set, e.g.
```
ot.ResourceMap_SetAsBool('GeneralLinearModelAlgorithmNormalizeData', True) 
```
is probably an error...

Moreover, the method ResourceMap_Set is unsafe because there is no mechanism of checking of the type/class of the given value.

From my point of view, it is necessary
+ to have a specific method ResourceMap_Add which declares a new entry and its type amongst scalar, boolean, string, ...
+ to suppress the method ResourceMap_Set to compel people to use ResourceMap_SetAs[Type].
 
Thanks.

**Comment by mbaudin47** [comment]
@mathieu-couplet is right : the `add` method would provide the feature that @jschueller mentions while providing the safeguard that I would like to have.

**Comment by mbaudin47** [comment]
This issue was fixed at #1342
